evm_arithmetization/src/prover.rs

```diff
@@ -249,6 +340,1098 @@ type ProofWithMemCaps<F, C, H, const D: usize> = (
     MerkleCap<F, H>,
 );
 
+/// Compute all STARK proofs. STARK-batching version.
+pub(crate) fn prove_with_traces_batch<F, P, C, const D: usize>(
+    all_stark: &AllStark<F, D>,
+    config: &StarkConfig,
+    trace_poly_values: [Vec<PolynomialValues<F>>; NUM_TABLES],
+    public_values: PublicValues,
+    timing: &mut TimingTree,
+    abort_signal: Option<Arc<AtomicBool>>,
+) -> Result<BatchStarkProofWithPublicInputs<F, C, D, NUM_TABLES>>
+where
+    F: RichField + Extendable<D>,
+    P: PackedField<Scalar = F>,
+    C: GenericConfig<D, F = F>,
+{
+    let rate_bits = config.fri_config.rate_bits;
+    let cap_height = config.fri_config.cap_height;
+
+    let trace_poly_values_sorted: [_; NUM_TABLES] = Table::all_sorted()
+        .iter()
+        .map(|&table| trace_poly_values[*table].clone())
+        .collect::<Vec<_>>()
+        .try_into()
+        .unwrap();
+
+    // We compute the Field Merkle Tree of all STARK traces.
+    let trace_polys_values_sorted_flat: Vec<_> = trace_poly_values_sorted
+        .clone()
+        .into_iter()
+        .flatten()
+        .collect();
+    let num_trace_polys = trace_polys_values_sorted_flat.len();
```

Comment:
Same here, the clones are way too heavy

Reply:
These clones are a bit trickier to remove, but I got rid of the other two sections.